### PR TITLE
Adding ceiling function

### PR DESCRIPTION
**Pull request by selwyni**
Added the ceiling function code, including the alias for CEIL and CEILING, in addition to a SQL test and a unit test for the utility function. Ceiling function has support for type Decimal, Integer, Big/Small/Tiny/Integer.

### CODE REVIEW

File: test/sql/decimal_functions_sql_test.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+class DecimalFunctionsSQLTest : public PelotonTest {};
+
+  TEST_F(DecimalFunctionsSQLTest, CeilTest) {
+    auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+    auto txn = txn_manager.BeginTransaction();
+    catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+    catalog::Catalog::GetInstance()->Bootstrap();
+    txn_manager.CommitTransaction(txn);
+    // Create a t
+    txn = txn_manager.BeginTransaction();
```
Comment (saatviks) [inline test/sql/decimal_functions_sql_test.cpp]: This BeginTransaction needs a corresponding EndTransaction else there's a memory leak - refer to my pull request. I had made the same mistake.

**Comment by apavlo** [comment]
@saatviks Your PR seems to now contain commits from other PRs. You may to merge your changes into a clean branch.

**Comment by saatviks** [comment]
@apavlo I'm sorry I dont understand. My PR #889 is already merged. This is another team's PR.

**Comment by hzxa21** [comment]
What is the status of this PR?

**Comment by selwyni** [comment]
Ready to merge

**Comment by pmenon** [comment]
Not yet, it has to be reviewed.

**Comment by selwyni** [comment]
I'll get on it, my bad

File: test/function/decimal_functions_test.cpp [hunk 2]
```diff
@@ -47,6 +47,59 @@ TEST_F(DecimalFunctionsTests, SqrtTest) {
...
+    EXPECT_FALSE(result.IsNull());
+    EXPECT_EQ(ceil(in), result.GetAs<double>());
+  }
+
+  args = {type::ValueFactory::GetNullValueByType(type::TypeId::DECIMAL)};
+  auto result = function::DecimalFunctions::_Ceil(args);
+  EXPECT_TRUE(result.IsNull());
+}
+
+TEST_F(DecimalFunctionsTests, CeilTestInt) {
```
Comment (selwyni) [inline test/function/decimal_functions_test.cpp]: I put my integer test functions for Ceiling under decimal_functions_test.cpp, should they be moved elsewhere?

Comment (pmenon) [reply test/function/decimal_functions_test.cpp]: Sure, that's fine.

**Comment by pmenon** [comment]
@selwyni I cleaned up the code a little bit. Can you run `clang-format` on the files you've changed here?

**Comment by hzxa21** [comment]
@selwyni It seems like the build failed because there are conflicts you haven't resolved (e.g. in decimal_type.cpp).

**Comment by hzxa21** [comment]
The ceil test was not passed. Could you look into that? @selwyni 
```
59: [0;32m[ RUN      ] [mDecimalFunctionsSQLTest.CeilTest
59: 2017-12-08 00:45:21 [test/sql/testing_sql_util.cpp:58:ExecuteSQLQuery] INFO  - Query: select id, ceil(income) from foo;
59: decimal_functions_sql_test: /home/travis/build/cmu-db/peloton/src/catalog/proc_catalog.cpp:129: std::unique_ptr<peloton::catalog::ProcCatalogObject> peloton::catalog::ProcCatalog::GetProcByName(const string&, const std::vector<peloton::type::TypeId>&, peloton::concurrency::Transaction*) const: Assertion `((*result_tiles)[0]->GetTupleCount() <= 1)' failed.
```

**Comment by apavlo** [comment]
I will merge this after the build passeds.

**Comment by apavlo** [comment]
@selwyni Can you rebase?

**Pull request merged**